WalletWasabi.Fluent/Models/Wallets/WalletCoinsModel.cs

```diff
@@ -52,6 +53,21 @@ public WalletCoinsModel(Wallet wallet, IWalletModel walletModel)
 
 	public IObservableCache<Pocket, LabelsArray> Pockets { get; }
 
+	public async Task UpdateExcludedCoinsFromCoinjoinAsync(ICoinModel[] coinsToExclude)
+	{
+		await Task.Run(() =>
+		{
+			// TODO: To keep models in sync with business objects. Should be automatic.
+			foreach (var coinModel in List.Items)
+			{
+				coinModel.IsExcludedFromCoinJoin = coinsToExclude.Any(x => x.IsSame(coinModel));
+			}
```

Comment:
This should be needed as the Models are subscribed to the SmartCoins's changes, and shout set this value... Going to continue the investigation tomorrow.

Reply:
@ichthus1604 @SuperJMN 

It seems like `CoinListViewModel` holds a different CoinModel object than we have in `WalletCoinsModel`. `IsExcludedFromCoinJoin` is properly set on `CoinModel` objects in `CoinListViewModel`, but when you check the same coin in `WalletCoinsModel`, you don't see those values set.

Reply:
Talked to @SuperJMN on Slack and he discovered the problem. It will be fixed in a separate PR.